### DISCUSSION TITLE
Error Rendering Component with JWPlayer Library

### DISCUSSION THREAD

**Discussion by hicath** [software-mansion/react-native-screens#1330] [Q&A]
Other library reference: https://github.com/chaimPaneth/react-native-jw-media-player

Basically we only use that library to play a video on our React Native project, the code is pretty simple, we do something like this: 

```
<View>
  <JWPlayer .../>
</View>
```

or

```
<ScrollView>
  <JWPlayer .../>
</ScrollView>
```

but we got this error: 

`Exception thrown while executing UI block: child view controller:<JWPlayerKit.JWPlayerViewController: 0x7f91c807ac00> should have parent view controller:<RNSScreen: 0x7f9168512950> but actual parent is:<UIViewController: 0x7f91aef1d920>`

just in case you need a full trace we see on the screen:

```
Exception thrown while executing UI block: child view controller:<JWPlayerKit.JWPlayerViewController: 0x7f91c807ac00> should have parent view controller:<RNSScreen: 0x7f9168512950> but actual parent is:<UIViewController: 0x7f91aef1d920>

__44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke
    RCTUIManager.m:1201
__44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke.498
__RCTExecuteOnMainQueue_block_invoke
_dispatch_call_block_and_release
_dispatch_client_callout
_dispatch_main_queue_callback_4CF
__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
__CFRunLoopRun
CFRunLoopRunSpecific
GSEventRunModal
-[UIApplication _run]
UIApplicationMain
main
start_sim
0x0
0x0
```

I know it's something we should ask to the original author, but we haven't received any feedback from them, so because the error says something related to `RNSScreen`  I think it might help us if you also can give us an idea what's wrong here under the hood with RNSScreen and JWPlayer, I put my hope high to get a solution I can implement on application level to tackle this issue.

Once again, any explanation and suggested solution would be appreciated! 

Thank you.

**Comment by tboba** [comment]
Hi @hicath, thanks for reporting about this! Does the problem still occur for you? If so, please report an issue in the issues tab with steps for minimal reproduction. Also, hierarchy of the screens would be helpful there - if you have an ability for running your app on Xcode, please attach the hierarchy from there 🙏 

Thanks!
